tests/integration/log-proxy-tester/src/charm.py

```diff
@@ -23,13 +23,24 @@ class LogProxyTesterCharm(CharmBase):
     def __init__(self, *args):
         super().__init__(*args)
 
+        self._enable_syslog = self.model.config.get("syslog")
+        self._enable_file_forwarding = self.model.config.get("file_forwarding")
+        self._log_files = ["/bin/driver.log"]
+
         self._log_proxy = LogProxyConsumer(
-            charm=self, log_files=["/bin/driver.log"], container_name="workload"
+            charm=self,
+            container_name="workload",
+            enable_syslog=self._enable_syslog,
+            log_files=self._log_files if self._enable_file_forwarding else [],
         )
         self.framework.observe(
             self._log_proxy.on.promtail_digest_error,
             self._promtail_error,
         )
+        self.framework.observe(
+            self._log_proxy.on.log_proxy_endpoint_joined,
+            self._promtail_success,
+        )
```

Comment:
Iiuc, `log_proxy_endpoint_joined` would be emitted in tandem with `relation-changed`.
If so,
1. Do we still need `log_proxy_endpoint_joined`?
2. If yes, then are we implicitly relying on `relation-changed` to fire before `log_proxy_endpoint_joined`?

Reply:
No. `log_proxy_endpoint_joined` is emitted _once promtail has a valid configuration_ and is running.

`relation-changed` can easily be emitted when `endpoints` change or `promtail_binary_url` is set, neither of which is a sufficient condition on its own to actually start/configure promtail.

This is not necessarily comitant with any given relation event (promtail may still be being fetched, `endpoints` may not be available for some reason, or other), and this patch overtly removes any implicit assumptions about what will or will not happen first, since relying on any `relation-` event to signal that the charm is "ready" does not necessarily indicate that `promtail` has a valid configuration, is fetched, or is running yet.

`log_proxy_endpoint_joined` is still needed, since it is essentially "`promtail` has been started successfully". It is the only signal to the consumer charm that it is _actually_ sending logs.